tachyon/math/polynomials/univariate/lagrange_interpolation.h

```diff
@@ -52,36 +52,58 @@ bool LagrangeInterpolate(
     CHECK(F::BatchInverseInPlaceSerial(chunk));
   });
 
-  std::vector<F> final_coeffs = base::CreateVector(points.size(), F::Zero());
-  for (size_t i = 0; i < points.size(); ++i) {
-    // See comments in |UnivariateDenseCoefficients::FromRoots()|.
-    // clang-format off
-    // NOTE(chokobole): This computes the polynomial whose roots are {x₀, ..., xᵢ₋₁, xᵢ₊₁, ..., xₙ₋₁}.
-    // Nᵢ(X) = (X - x₀)...(X - xᵢ₋₁)(X - xᵢ₊₁)...(X - xₙ₋₁)
-    //       = cₙ₋₁Xⁿ⁻¹ + cₙ₋₂Xⁿ⁻² + ... + c₁X¹ + c₀X⁰
-    // |coeffs[i]| = cᵢ
-    // clang-format on
-    std::vector<F> coeffs = base::CreateVector(points.size(), F::Zero());
-    coeffs[0] = F::One();
-    size_t k_start = 1;
-    for (size_t j = 0; j < points.size(); ++j) {
-      if (i == j) continue;
-      for (size_t k = k_start; k > 0; --k) {
-        coeffs[k] = coeffs[k - 1] - points[j] * coeffs[k];
-      }
-      ++k_start;
-      coeffs[0] *= -points[j];
-    }
+  std::vector<std::vector<F>> coeffs_sums = base::ParallelizeMap(
+      evals, [&points, &denoms](absl::Span<const F> chunk, size_t chunk_offset,
+                                size_t chunk_size) {
+        // See comments in |UnivariateDenseCoefficients::FromRoots()|.
+        // clang-format off
+        // NOTE(chokobole): This computes the polynomial whose roots are {x₀, ..., xᵢ₋₁, xᵢ₊₁, ..., xₙ₋₁}.
```

Comment:
does it compute xᵢ?

Reply:
Nope. it skips. Please check the line 83.

```c++
if (i == j) continue;
```